example/timer/odp_timer_accuracy.c

```diff
@@ -446,23 +511,38 @@ static int start_timers(test_global_t *test_global)
 	test_global->start_ns = start_ns;
 	test_global->period_tick = odp_timer_ns_to_tick(timer_pool, period_ns);
 
-	/* When mode is 1, set only one burst of timers initially */
-	if (mode)
+	/* When mode is not one-shot, set only one burst of timers initially */
+	if (mode != MODE_ONESHOT)
 		num_tmo = 1;
 
 	for (i = 0; i < num_tmo; i++) {
 		for (j = 0; j < burst; j++) {
 			timer_ctx_t *ctx = &test_global->timer_ctx[idx];
 			odp_timer_start_t start_param;
 
-			nsec = offset_ns + (i * period_ns) + (j * burst_gap);
-			ctx->nsec = start_ns + nsec;
-
-			start_param.tick_type = ODP_TIMER_TICK_ABS;
-			start_param.tick = start_tick + odp_timer_ns_to_tick(timer_pool, nsec);
-			start_param.tmo_ev = ctx->event;
-
-			ret = odp_timer_start(ctx->timer, &start_param);
+			if (mode == MODE_PERIODIC) {
+				odp_timer_periodic_start_t start_param;
+
+				/*
+				 * Burst is centered at the middle of the period. If the burst
+				 * doesn't fit in the period, odp_timer_periodic_start() fails.
+				 */
+				nsec = period_ns / 2 - burst_gap * burst / 2 + j * burst_gap;
+				ctx->nsec = start_ns + nsec;
+				start_param.freq_multiplier = test_global->opt.multiplier;
+				start_param.first_tick =
+					start_tick + odp_timer_ns_to_tick(timer_pool, nsec);
+				start_param.tmo_ev = ctx->event;
```

Comment:
This seems quite complex. Would it be possible to simply set `start_param.first_tick`to zero if `burst_gap` was not given? And if given, e.g. `nsec = burst_gap * (burst + 1)`.

Reply:
> Would it be possible to simply set `start_param.first_tick`to zero if `burst_gap` was not given?

That should work.

> And if given, e.g. `nsec = burst_gap * (burst + 1)`.

Maybe you mean `nsec = burst_gap * (j + 1)`. Regardless, that's an automatic TOO_NEAR with the default burst_gap of 0. The complexity is there to minimize the probability of TOO_NEAR and TOO_FAR.

Reply:
Yep, meant `nsec = burst_gap * (j + 1)`, but only if `burst_gap` has been given by user (!=0). This way it would be user's responsibility to pick a large enough value.

Reply:
In v3, changed this to something that's closer to the non-periodic case, using offset_ns, but if nsec is 0, then first_tick is 0.